From 4221c07f2c1db32ffa44eed68bd7788ade2e4a9b Mon Sep 17 00:00:00 2001
From: Meng Dongyang <daniel.meng@rock-chips.com>
Date: Tue, 11 Oct 2016 17:25:14 +0800
Subject: [PATCH] power: rk818_charger: get cable state when usb charge
 function init

RK818 will miss the notify of charge type changing because
the charge cable state is init when u2phy probe but rk818 probe after
u2phy. So we need to get the charge cable state when rk818 probe.

Change-Id: I3682d764ae3f9a56a1ba85ba8b81ea7f1aacdf49
Signed-off-by: Meng Dongyang <daniel.meng@rock-chips.com>
---
 drivers/power/rk818_charger.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/power/rk818_charger.c b/drivers/power/rk818_charger.c
index 832f115f9bf7..aeff14f20498 100644
--- a/drivers/power/rk818_charger.c
+++ b/drivers/power/rk818_charger.c
@@ -1060,6 +1060,8 @@ static long rk818_cg_init_usb(struct rk818_charger *cg)
 
 		cg->cable_edev = edev;
 
+		schedule_delayed_work(&cg->host_work, 0);
+
 		CG_INFO("register typec extcon evt notifier\n");
 	} else {
 		INIT_DELAYED_WORK(&cg->usb_work, rk818_cg_bc_evt_worker);
-- 
2.35.3

